src/Storage/Storage.Management/StorageAccount/NewAzureStorageAccount.cs

```diff
@@ -653,151 +654,189 @@ public override void ExecuteCmdlet()
                 {
-                    createParameters.Encryption.EncryptionIdentity = new EncryptionIdentity();
-                    createParameters.Encryption.EncryptionIdentity.EncryptionUserAssignedIdentity = this.KeyVaultUserAssignedIdentityId;
+                    createContent.Encryption.EncryptionIdentity = new Track2Models.EncryptionIdentity
+                    {
+                        EncryptionUserAssignedIdentity = this.KeyVaultUserAssignedIdentityId
+                    };
+
                 }
             }
             if (this.minimumTlsVersion != null)
             {
-                createParameters.MinimumTlsVersion = this.minimumTlsVersion;
+                createContent.MinimumTlsVersion = new Track2Models.MinimumTlsVersion(this.minimumTlsVersion);
             }
             if (this.allowBlobPublicAccess != null)
             {
-                createParameters.AllowBlobPublicAccess = this.allowBlobPublicAccess;
+                createContent.AllowBlobPublicAccess = this.allowBlobPublicAccess;
             }
             if (this.RoutingChoice != null || this.publishMicrosoftEndpoint != null || this.publishInternetEndpoint != null)
             {
-                createParameters.RoutingPreference = new RoutingPreference(this.RoutingChoice, this.publishMicrosoftEndpoint, this.publishInternetEndpoint);
+                createContent.RoutingPreference = new Track2Models.RoutingPreference
+                {
+                    RoutingChoice = new Track2Models.RoutingChoice(this.RoutingChoice),
+                    PublishMicrosoftEndpoints = this.publishMicrosoftEndpoint,
+                    PublishInternetEndpoints = this.publishInternetEndpoint
+                };
             }
             if (allowSharedKeyAccess != null)
             {
-                createParameters.AllowSharedKeyAccess = allowSharedKeyAccess;
+                createContent.AllowSharedKeyAccess = this.allowSharedKeyAccess;
             }
             if (enableNfsV3 != null)
             {
-                createParameters.EnableNfsV3 = enableNfsV3;
+                createContent.EnableNfsV3 = this.enableNfsV3;
             }
             if (this.EdgeZone != null)
             {
-                createParameters.ExtendedLocation = new ExtendedLocation()
+                createContent.ExtendedLocation = new Track2Models.ExtendedLocation
                 {
-                    Type = ExtendedLocationTypes.EdgeZone,
-                    Name = this.EdgeZone
+                    Name = this.EdgeZone,
+                    ExtendedLocationType = Track2Models.ExtendedLocationTypes.EdgeZone,
                 };
             }
             if (sasExpirationPeriod != null)
             {
-                createParameters.SasPolicy = new SasPolicy(sasExpirationPeriod.Value.ToString(@"d\.hh\:mm\:ss"));
+                createContent.SasPolicy = new Track2Models.SasPolicy(sasExpirationPeriod.Value.ToString(@"d\.hh\:mm\:ss"), null);
```

Comment:
This line will throw exception: 
System.ArgumentNullException
  HResult=0x80004003
  Message=Value cannot be null.
Parameter name: value
  Source=Azure.ResourceManager.Storage
  StackTrace:
   at Azure.ResourceManager.Storage.Models.ExpirationAction..ctor(String value)
   at Azure.ResourceManager.Storage.Models.ExpirationAction.op_Implicit(String value)
   at Microsoft.Azure.Commands.Management.Storage.NewAzureStorageAccountCommand.ExecuteCmdlet() in D:\code\PSH_Dev\src\Storage\Storage.Management\StorageAccount\NewAzureStorageAccount.cs:line 795
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()

Following code works in my testing.
`new Track2Models.SasPolicy(sasExpirationPeriod.Value.ToString(@"d\.hh\:mm\:ss"), Track2Models.ExpirationAction.Log)`

Reply:
Fixed